app/portal/chapters/page.js

```diff
@@ -85,7 +100,11 @@ export default function Page() {
             <div style={{ height: 500, width: "100%" }}>
               <DataGrid
                 rows={rows}
-                columns={columns}
+                columns={columns.map((column) =>
+                  column.field === "chapterName"
+                    ? { ...column, renderCell: handleRowClick }
```

Comment:
This is a cool way to ensure you're applying the `handleRowClick` function to the proper column, love it.